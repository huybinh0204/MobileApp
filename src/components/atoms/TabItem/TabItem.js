import PropTypes from 'prop-types';
import React from 'react';
import { TouchableWithoutFeedback } from 'react-native';
import IconSvg from '_components/atoms/IconSvg/IconSvg';
import { normalize } from '_utilities/screen';
import { TabItemContainer, TabItemLabel } from './TabItem.styles';

const TabItem = ({ onPress, title, icon, isActive, testID }) => {
  return (
    <TouchableWithoutFeedback
      accessibilityLabel={`${title} Tab Button`}
      testID={testID}
      onPress={onPress}
    >
      <TabItemContainer>
        <IconSvg testID="tabIcon" icon={icon} width={normalize(28)} height={normalize(28)} />
        <TabItemLabel isActive={isActive} testID="tabLabel">
          {title}
        </TabItemLabel>
      </TabItemContainer>
    </TouchableWithoutFeedback>
  );
};

TabItem.defaultProps = {
  testID: 'tabItem',
};

TabItem.protoTypes = PropTypes.shape({
  onPress: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  isActive: PropTypes.bool.isRequired,
  testID: PropTypes.string,
});

export default TabItem;
