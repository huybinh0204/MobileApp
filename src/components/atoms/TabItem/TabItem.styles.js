import styled from '@emotion/native';
import BaseText from '_components/atoms/BaseText/BaseText';

export const TabItemContainer = styled.View({
  flex: 1,
  alignItems: 'center',
  justifyContent: 'center',
});

export const TabItemLabel = styled(BaseText)(({ theme, isActive }) => ({
  ...theme.typography.tinyContentBold,
  color: isActive ? theme.colors.alpha500 : theme.colors.beta100,
  marginTop: 5,
  textAlign: 'center',
}));
