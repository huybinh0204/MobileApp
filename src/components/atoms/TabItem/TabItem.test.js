import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import TabItem from './TabItem';

describe('TabItem', () => {
  let props;
  let onPressMock = jest.fn();

  beforeEach(() => {
    props = {
      icon: 'ahead',
      isActive: true,
      onPress: onPressMock,
      title: 'test',
    };
  });

  it('should match the snapshot', () => {
    const { toJSON } = render(
      <ComponentWithProviders component={TabItem} componentProps={props} />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should have a icon', () => {
    const { getByTestId } = render(
      <ComponentWithProviders component={TabItem} componentProps={props} />
    );

    const tabIcon = getByTestId('tabIcon');

    expect(tabIcon).toBeTruthy();
  });

  it('should have a label', () => {
    const { getByTestId } = render(
      <ComponentWithProviders component={TabItem} componentProps={props} />
    );

    const tabItem = getByTestId('tabLabel');

    expect(tabItem).toHaveTextContent('test');
  });

  it('should trigger onPress callback', () => {
    const { getByA11yLabel } = render(
      <ComponentWithProviders component={TabItem} componentProps={props} />
    );

    const tabItem = getByA11yLabel('test Tab Button');

    fireEvent.press(tabItem);

    expect(onPressMock).toHaveBeenCalledTimes(1);
  });
});
