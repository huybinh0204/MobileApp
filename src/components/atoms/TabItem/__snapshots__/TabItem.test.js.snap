// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabItem should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessibilityLabel="test Tab Button"
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "flex": 1,
                              "justifyContent": "center",
                            },
                            undefined,
                          ]
                        }
                        testID="tabItem"
                      >
                        <Text
                          allowFontScaling={false}
                          testID="tabIcon"
                        >
                          not Found
                        </Text>
                        <Text
                          allowFontScaling={false}
                          isActive={true}
                          style={
                            Array [
                              Object {
                                "color": "#3D3397",
                                "fontFamily": "Poppins-SemiBold",
                                "fontSize": 21.5,
                                "lineHeight": 30.5,
                                "marginTop": 5,
                                "textAlign": "center",
                              },
                              undefined,
                            ]
                          }
                          testID="tabLabel"
                        >
                          test
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
